pallets/cash/src/internal/initialize.rs

```diff
@@ -115,12 +115,17 @@ pub fn initialize_block<T: Config>(now: Timestamp) -> Result<(), Reason> {
         }
     }
 
-    <Module<T>>::deposit_event(Event::TransferCash(
-        GATEWAY_VOID,
-        last_miner,
-        last_miner_share_principal,
-        cash_index_new,
-    ));
+    if last_miner_share_principal != CashPrincipalAmount::ZERO {
+        // Filter Transfer 0 events emitted when miners are paid
+        // These are just cluttering up tools with noise
+        <Module<T>>::deposit_event(Event::TransferCash(
+            GATEWAY_VOID,
+            last_miner,
+            last_miner_share_principal,
+            cash_index_new,
+        ));
+    }
+
     <Module<T>>::deposit_event(Event::MinerPaid(last_miner, last_miner_share_principal));
```

Comment:
Should we filter this one too?

Reply:
Yeah i wasn't sure on this one that is not what was described in the ticket. I think it probably makes sense to have one of these per block, something like our version of a coinbase tx event